src/tests/array-at.ts

```diff
@@ -1,42 +1,64 @@
 import { doNotExecute, Equal, Expect } from "./utils";
 
 doNotExecute(async () => {
-  const arr = [false, 1, '2'] as const;
+  const arr = [false, 1, "2"] as const;
 
-  const a = arr.at(0)
-  const b = arr.at(1)
-  const c = arr.at(2)
-  const d = arr.at(3)
+  const a = arr.at(0);
+  const b = arr.at(1);
+  const c = arr.at(2);
+  const d = arr.at(3);
+  const e = arr.at(1.5);
   type tests = [
     Expect<Equal<typeof a, false>>,
     Expect<Equal<typeof b, 1>>,
-    Expect<Equal<typeof c, '2'>>,
+    Expect<Equal<typeof c, "2">>,
     Expect<Equal<typeof d, undefined>>,
-  ]
+    Expect<Equal<typeof e, 1>>,
+  ];
 });
 
 doNotExecute(async () => {
-  const arr = [false, 1, '2'] as const;
+  const arr = [false, 1, "2"] as const;
 
-  const a = arr.at(-1)
-  const b = arr.at(-2)
-  const c = arr.at(-3)
-  const d = arr.at(-4)
+  const a = arr.at(-1);
+  const b = arr.at(-2);
+  const c = arr.at(-3);
+  const d = arr.at(-4);
+  const e = arr.at(-1.5);
   type tests = [
-    Expect<Equal<typeof a, '2'>>,
+    Expect<Equal<typeof a, "2">>,
     Expect<Equal<typeof b, 1>>,
     Expect<Equal<typeof c, false>>,
     Expect<Equal<typeof d, undefined>>,
-  ]
+    Expect<Equal<typeof e, "2">>,
+  ];
 });
 
 doNotExecute(async () => {
-  const arr = [false, 1, '2'] as const;
-  const index: number = 1
-  const a = arr.at(index)
-  // WARN: with `"strictNullChecks": true,` the correct type here should include `undefined`
-  // but the current implementation does not type this as including `undefined`
-  type tests = [
-    Expect<Equal<typeof a, false | 1 | "2">>,
-  ]
-});
\ No newline at end of file
+  const arr = [false, 1, "2"] as const;
+
+  function index() {
+    return Math.random() > 0.5 ? 0 : 1;
+  }
+
+  const a = arr.at(index());
```

Comment:
Wouldn't this be simpler with a forced type like this?
```suggestion
  const index = 1 as 1 | 2;
  const a = arr.at(index);
```

Or is it done like this for "expressivity" of the tests (like saying "sometimes we don't know the type in advance" but with code)?